src/server/session.ts

```diff
@@ -1738,9 +1738,22 @@ namespace ts.server {
         }
 
         private getEditsForFileRename(args: protocol.GetEditsForFileRenameRequestArgs, simplifiedResult: boolean): ReadonlyArray<protocol.FileCodeEdits> | ReadonlyArray<FileTextChanges> {
-            const { file, project } = this.getFileAndProject(args);
-            const changes = project.getLanguageService().getEditsForFileRename(toNormalizedPath(args.oldFilePath), toNormalizedPath(args.newFilePath), this.getFormatOptions(file), this.getPreferences(file));
-            return simplifiedResult ? this.mapTextChangesToCodeEdits(project, changes) : changes;
+            const oldPath = toNormalizedPath(args.oldFilePath);
+            const newPath = toNormalizedPath(args.newFilePath);
+            const formatOptions = this.getHostFormatOptions();
+            const preferences = this.getHostPreferences();
+
+            const changes: (protocol.FileCodeEdits | FileTextChanges)[] = [];
+            this.projectService.forEachProject(project => {
+                for (const fileTextChanges of project.getLanguageService().getEditsForFileRename(oldPath, newPath, formatOptions, preferences)) {

```

Comment:
Do you need to exclude orphan projects or the configured projects which have no open file?

Reply:
Also the projects with language service disabled?

Reply:
If the project has no files, `getEditsForFileRename` should just return nothing. Should `getLanguageService()` throw an error if the language service is disabled? Maybe we should have `tryGetLanguageService()` that returns `LanguageService | undefined`?

Reply:
getLanguageSevice will return language service all the time but it will be disabled (cleared semantic cache. I am working on changes where language service itself will know that its disabled but for now this info is in project and we should ignore those projects.)
For orphan projects, we are going through them unnecessarily no? we could avoid all that work by just ignoring these projects

Reply:
OK, we will now skip those projects.